libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/MapboxNavigationAccounts.kt

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.navigation.core.accounts
+
+import com.mapbox.navigation.base.internal.accounts.UrlSkuTokenProvider
+
+/**
+ * This class generates and retains the Navigation SDK's SKU token according to internal Mapbox policies
+ */
+internal object MapboxNavigationAccounts : UrlSkuTokenProvider {
+
+    private const val SKU_KEY = "sku"
+
+    /**
+     * Returns a token attached to the URL query or the given [resourceUrl].
+     */
+    @Synchronized
+    override fun obtainUrlWithSkuToken(resourceUrl: String, querySize: Int): String {
```

Comment:
- why need `@Synchronized`?
- I would replace this method with just `fun obtainUrlWithSkuToken(resourceUrl: String)` and working with it via `java.net.URL` or `java.net.URI` APIs(it's libraries so we need fewer tests)

Reply:
> why need @Synchronized?

Good catch, a carryover from old implementation, I don't think we need it.

I also refactored the method to use the `URI`/`URL` wrappers instead of working on raw strings

Reply:
Cc @abhishek1508 for extra historical context on `@Synchronized`